### ISSUE TITLE
A fully automated release process

### ISSUE THREAD

**Issue by thomaseizinger** [comit-network/comit-rs#1476]
## Goal:
Have a fully automated release process

## Problem:
Currently, releasing a new version involves several manual steps:

- branching off locally
- updating the version of cnd
- making a commit
- tagging the commit
- pushing the release branch

## Recommendation:

> To make the process more smooth, this assumes GitFlow because of the [limitation](https://github.com/comit-network/comit-rs/issues/1382#issuecomment-533947701) regarding release branches of our current branching model that was mentioned in the last discussion.

Ideally, we would only specify, what the next version that we want to release is and some automation handles the rest. I suggest the following process:

1. The triggering action of creating a new release is to open an issue that has a specific title like: "Release version 0.3.0"
- has to be limited to a certain set of users
2. Have a GitHub action that listens for the creation of a new issue and does the following:
- Parse the new version from the issue title
- Checkout develop branch
- Create a new branch `release/X.Y.Z`
- Update the version of `cnd` to `X.Y.Z`
- Move the Changelog entry from "Unreleased" to the new version
- Make a commit
- Push the release branch
- Request review from the team
3. Configure mergify to merge release branches upon N number of approvals
4. Trigger a GitHub action when a release branch is merged that:
- tags the commit with the given version number
- create a release on GitHub and copy the relevant changelog section to the release
- create release build for necessary platforms and attaches it to the release
- close the ticket once the PR is merged (open the PR with `Fixes XYZ?`)

PROFIT!

**Comment by thomaseizinger** [comment]
At the moment, releasing doesn't seem to take up too much of our time. I would like to invite everyone (@D4nte, @bonomat, @luckysori, @tcharding, @da-kami) to track the amount of time they spend on releasing in this ticket. Just edit this comment please:

@D4nte: 0 minutes
@bonomat: 0 minutes
@luckysori: 0 minutes
@tcharding: 0 minutes
@da-kami: 0 minutes
@thomaseizinger: 0 minutes

The action proposed in this ticket would be reusable across all repositories. Hence, the time tracking here applies to ALL our repositories.
